.github/workflows/staging.yml

```diff
@@ -0,0 +1,201 @@
+        id: cache-node-modules
+        with:
+          path: |
+            node_modules
+            functions/node_modules
+          key: node-modules-${{ hashFiles('**/yarn.lock') }}-${{ env.CACHE_VERSION }}
+
+      - name: Run accessibility tests
+        run: yarn test:a11y
+
+  test-unit:
+    name: Test – Unit
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@master
+
+      - name: Get cached dependencies
+        uses: actions/cache@v2
+        id: cache-node-modules
+        with:
+          path: |
+            node_modules
+            functions/node_modules
+          key: node-modules-${{ hashFiles('**/yarn.lock') }}-${{ env.CACHE_VERSION }}
+
+      - name: Run unit tests
+        run: yarn test:unit
+
+  deploy-hosting:
+    name: Deploy – Hosting
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@master
+
+      - name: Get cached build
+        id: cache-build
+        uses: actions/cache@v2
+        with:
+          path: |
+            functions/lib
+            public
+            firebase.json
+          key: build-${{ hashFiles('content/**', 'firebase/**', 'functions/src/**', 'src/**', 'static/**') }}-${{ env.CACHE_VERSION }}
+
+      - name: Get Firebase cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            .cache
+          key: firebase-${{ github.event.number }}-${{ env.CACHE_VERSION }}
+
+      - name: Deploy Firebase preview channel
+        id: deploy-preview-channel
+        uses: FirebaseExtended/action-hosting-deploy@v0
+        with:
+          repoToken: "${{ secrets.GITHUB_TOKEN }}"
+          firebaseServiceAccount: "${{ env.FIREBASE_SERVICE_ACCOUNT }}"
+          expires: 7d
+          projectId: "${{ env.FIREBASE_PROJECT_ID }}"
+          channelId: "${{ env.FIREBASE_CHANNEL_ID }}"
+        env:
+          FIREBASE_CLI_PREVIEWS: hostingchannels
+
+      - name: Update robots and sitemap files for staging environments
+        env:
+          FIND_AND_REPLACE: s|https://dev.maxmind.com|${{ steps.deploy-preview-channel.outputs.details_url }}|g
+        run: |
+          sed -i "s/Allow:/Disallow:/g" public/robots.txt
+          sed -i "$FIND_AND_REPLACE" public/robots.txt
+          sed -i "$FIND_AND_REPLACE" public/sitemap.xml
+          sed -i "$FIND_AND_REPLACE" public/sitemap-pages.xml
+
+      - name: Deploy updated files staging environment files
+        uses: FirebaseExtended/action-hosting-deploy@v0
+        with:
+          repoToken: "${{ secrets.GITHUB_TOKEN }}"
+          firebaseServiceAccount: "${{ env.FIREBASE_SERVICE_ACCOUNT }}"
+          expires: 30d
+          projectId: "${{ env.FIREBASE_PROJECT_ID }}"
+          channelId: "${{ env.FIREBASE_CHANNEL_ID }}"
+        env:
+          FIREBASE_CLI_PREVIEWS: hostingchannels
+
+  deploy-functions:
+    name: Deploy – Functions
+    needs: build
```

Comment:
How come we the `needs` value is different here than from production?

Reply:
I was thinking that the deploy jobs in production make sure that tests, if they run, successfully pass before deploying. In staging, I think it is fine if the testing and deployment happen concurrently to speed up the action.